### PR TITLE
More RxSwift changes

### PR DESCRIPTION
**Pull request by ashfurrow**
Just want to get these merged into develop before too long. Still more to do!

Also, no rush at all!

### CODE REVIEW

File: Kiosk/Bid Fulfillment/Models/NewUser.swift [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-    dynamic var creditCardToken: String?
-    dynamic var creditCardName: String?
-    dynamic var creditCardType: String?
-    dynamic var zipCode: String?
-    dynamic var name: String?
+import RxSwift
 
-    dynamic var hasBeenRegistered: Bool = false
+class NewUser {
+    var email = Variable<String?>(nil)
```
Comment (orta) [inline Kiosk/Bid Fulfillment/Models/NewUser.swift]: the the `nil` here a default value?

File: Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift [hunk 2]
```diff
@@ -33,16 +39,27 @@ class ConfirmYourBidViewController: UIViewController {
         let attrTitle = NSAttributedString(string: titleString, attributes:attributes)
         useArtsyLoginButton.setAttributedTitle(attrTitle, forState:useArtsyLoginButton.state)
 
-        RAC(self, "number") <~ numberSignal
+        numberSignal
+            .bindTo(_number)
+            .addDisposableTo(rx_disposeBag)
```
Comment (orta) [inline Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift]: this pattern may be possible turn into an NSObject extension?

File: Kiosk/Bid Fulfillment/Models/NewUser.swift (hunk 1 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/Models/NewUser.swift]: Correct, it's the initial value of the variable.

File: Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift (hunk 2 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift]: Yeah, that's a good idea. I've thought about how to get rid of at least the last line, but it's hard. I either need to pass in the dispose bag or a reference to `self`. Something to keep an eye out for a better abstraction.

File: Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift [hunk 3]
```diff
@@ -51,8 +68,9 @@ class ConfirmYourBidViewController: UIViewController {
         let auctionID = nav.auctionID
         
         let numberIsZeroLengthSignal = numberSignal.map(isZeroLengthString)
-        enterButton.rac_command = RACCommand(enabled: numberIsZeroLengthSignal.not()) { [weak self] _ in
-            guard let me = self else { return RACSignal.empty() }
+
+        enterButton.rx_action = CocoaAction(enabledIf: numberIsZeroLengthSignal.not(), workFactory: { [weak self] _ in
```
Comment (orta) [inline Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift]: could you ditch the `workFactory:` here and use the trailing block syntax?

File: Kiosk/Bid Fulfillment/KeypadViewModel.swift [hunk 4]
```diff
@@ -1,84 +1,78 @@
...
-    lazy var stringValueSignal: RACSignal = {
-        RACObserve(self, "stringValue")
-    }()
+    lazy var stringValue = Variable("")
     
-    // MARK: - Commands
+    // MARK: - Actions
     
-    // I have no idea why, but if you try and use `[weak self]` in the closure definition of a RACCommand, the compiler segfaults ¯\_(ツ)_/¯
+    // I have no idea why, but if you try and use `[weak self]` in the closure definition of a Action, the compiler segfaults ¯\_(ツ)_/¯
```
Comment (orta) [inline Kiosk/Bid Fulfillment/KeypadViewModel.swift]: did you test this against the RxSwift?

Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/KeypadViewModel.swift]: It was a compiler thing, but that was with Swift 1.0. Definitely worth trying again.

File: Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift (hunk 3 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/ConfirmYourBidViewController.swift]: Yes definitely, but I want to get people using the Action library and need some explicit code examples :wink:

File: Kiosk/Bid Fulfillment/BidCheckingNetworkModel.swift [hunk 5]
```diff
@@ -22,103 +26,127 @@ class BidCheckingNetworkModel: NSObject {
...
-                for position in newBidderPositions! {
-                    if position.highestBid?.id == topBidID {
-                        self?.isHighestBidder = true
+    private func checkForMaxBid() -> Observable<Void> {
+        return getMyBidderPositions()
+            .doOnNext{ [weak self] newBidderPositions -> Void in
+                guard let me = self else { return }
+
+                if let topBidID = me.mostRecentSaleArtwork?.saleHighestBid?.id {
+                    for position in newBidderPositions {
```
Comment (orta) [inline Kiosk/Bid Fulfillment/BidCheckingNetworkModel.swift]: `for ... where`

**Comment by orta** [comment]
Alright, let's get this in before it becomes a mega-merge. I've given it a third-over and I can't see anything particularly off

File: Kiosk/Bid Fulfillment/BidCheckingNetworkModel.swift (hunk 5 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/BidCheckingNetworkModel.swift]: :tada:

**Pull request merged**